### PR TITLE
Titanium and Ti-Alloy (Part 1: Materials and basic tools)

### PR DESCRIPTION
**Pull request by gadgetpatch**
#### Summary
Content "Adds a durable, lightweight metal alternative to steel, via titanium and its alloys."

#### Purpose of change
Agile characters have fewer options for good mid-to-late-game armors or weapons. In the same way that adding more varieties of real-world steel crafting opened up options for crafty characters, this PR introduces some real world titanium tools, weapons, and armors for dexterous ones. Titanium gear isn't meant to completely overshadow performance steel, but instead to offer low encumbrance armor and high-speed melee weapon options for high-mobility characters. In exchange, it's generally a bit less protective than equivalent steel, and harder to find and maintain.

#### Describe the solution
Included in this PR are several small blades (both gimmicky ones and useful tools), cut-resistant alloy chain armor, dual-hardness alloy ballistic plates, and mundane titanium items like durable jewelry and cutlery that survivors can find by scavenging, all based on researched real-world examples. This PR will also serve as the foundation for future work in new equipment crafting options, better simulated inert-gas welding, light but sturdy land and water vehicles, more realistic future aircraft bodies, and so on.

Generally speaking, titanium weapons are non-conductive, faster and easier to swing, and deal more damage-per-stamina than equivalent steel; however, they deal similar damage-per-hit, and are largely only useful for bladed weapons. Tools and vehicle frames generally perform better than medium steel equivalents, and are both lighter and more durable, but harder to find. Future PRs will focus on crafting and disassembly, titanium hardening and forging, fluxless inert gas welding, titanium-bodied vehicles, and fine-tuning stats for game balance.

For our survivor's purposes, Titanium comes in 2 main categories: pure titanium, and titanium-aluminum alloy:
- **Titanium** (based on [Grade 4 pure titanium](https://www.matweb.com/search/DataSheet.aspx?MatGUID=4b86c47a545247afae3da55d62381f89), often used for easy workability, and low cost applications):
55% the weight and encumbrance of low carbon steel, and has better environmental resistance than standard steel. In exchange, it's less common, has lower impact defense and needs more maintenance than standard steel, and is outperformed by high carbon steel in everything but light weight. Something of a middle ground between steel and aluminum. that only takes average metalworking skill to work, and still makes decent tools.
> _In a future PR, titanium can be chemically hardened with nitrogen gas to form "Hardened Titanium" (aka Titanium Nitride); roughly equivalent to case-hardened steel, but requires an Arc Furnace, and can't be repaired by welding or forging._

- **Titanium Alloy** (based on the [Grade 5 "workhorse" alloy](https://www.matweb.com/search/DataSheet.aspx?MatGUID=a0655d261898456b958e5f825ae85390) used in ultralight vehicles and wear-resistant tools):
Generally speaking, titanium alloy is rarer and harder to work than standard steel, but is roughly on par with high-carbon steels; it trades lower impact defense for better environmental and ballistic resistance, and equivalent cut resistance. It's also still 55% of the weight and encumbrance of carbon steel.
> _In a future PR, it can be quench tempered to form "Tempered Titanium" (aka martensitized α-prime-β Titanium Alloy); superior cut and ballistic protection than even tempered steel, but requires both high skill and an Arc Furnace._

#### Describe alternatives you've considered
Using Aluminum for similar purposes, splitting aluminum up into can-grade aluminum and alloy-grade aluminum that's competitive with low-grade steel, and make superalloy craftable (if I could find a real-world aluminum alloy to base it on).

#### Testing
Spawn group testing to check for reasonable spawn rates for new additions to existing item groups. Basic combat A/B testing of titanium weapons and armor against standard zombies and TALONs/turrets, on a low skill 12 Dexterity character, and comparing it to equivalent steel gear on a 12 Strength character.

#### Additional context
ChainmailComparison
KnifeScreen
TitaniumStats

### CODE REVIEW

**Comment by Aerin-of-the-Toast** [comment]
I don't know much about material sciences but the titanium weapons (mostly the katana and to a lesser extent the wakizashi) shouldn't be doing the same damage as their steel counterparts. 
Since they're far lighter - about half the weight - ingame this already translates to a faster attack speed and better stamina efficiency. Which is fine, but it does mean it has a large DPS increase. But the main issue is that by making the blade so much lighter you affect the balance and cutting power of the sword as a whole - the balance is going to be more towards the hilt, which makes the sword more agile but generally worse at raw chopping power. Similarly lower overall weight also means less momentum on the swings. 
Think of it like how an axe has a lot of chopping power due to being heavier and the weight being more top-heavy, but in reverse.

**Comment by gadgetpatch** [comment]
> I don't know much about material sciences but the titanium weapons (mostly the katana and to a lesser extent the wakizashi) shouldn't be doing the same damage as their steel counterparts. Since they're far lighter - about half the weight - ingame this already translates to a faster attack speed and better stamina efficiency. Which is fine, but it does mean it has a large DPS increase. But the main issue is that by making the blade so much lighter you affect the balance and cutting power of the sword as a whole - the balance is going to be more towards the hilt, which makes the sword more agile but generally worse at raw chopping power. Similarly lower overall weight also means less momentum on the swings. Think of it like how an axe has a lot of chopping power due to being heavier and the weight being more top-heavy, but in reverse.

I agree generally on tuning the numbers down. I'm happy with the armor right now, but the swords especially were rough ballparks, while I got all the content roughed out. I've been lucky enough to find some military studies on titanium armors, but hard data on titanium swords is tough to come by. There's not a lot of formal laboratory studies on it, believe it or not!

That said, I've read anecdotes from a [handful of titanium smiths](https://www.madscienceforge.com/laboratory/pages/en/faq.php) who make machetes and swords, who claim the stiffness of titanium prevents flexing on strikes, which makes up for (and in some cases improves) long blade strike performance, _despite_ the low mass. But the best titanium swords that are built for chopping instead of quick cuts do require changes to the blade shape, mostly to tune the balance like you say.

I'm still not sure how I want to address that part. It might be interesting to let survivors invent fully original blade shapes for the lighter metal... but I'd just as soon assume that at Fabrication 8+ they're able to adjust the design on classic blades, but still call it a katana, say.

**Review by kevingranade — CHANGES_REQUESTED** [review]
A couple of issues here.
1. Too many changes. I'm seeing a lot of stuff bundled in here like new plate carriers that have nothing to do with titanium.  Those can go into a seperate PR and most likely merge quickly.
2. We discussed swords, you got a hard no on swords, you said you were dropping the swords, they should not be in this pr.

**Comment by gadgetpatch** [comment]
>     1. Too many changes. I'm seeing a lot of stuff bundled in here like new plate carriers that have nothing to do with titanium.  Those can go into a seperate PR and most likely merge quickly.

I was adding missing steel ESAPI side plates while tuning steel values, and adding support for dual-hardness titanium side plates. It made sense to me wrap them together as part of the armor overhaul, but I can def split the steel plates out if that's a general preference.
 
>     2. We discussed swords, you got a hard no on swords, you said you were dropping the swords, they should not be in this pr.

From what I remember, the concern was that we hadn't found sources for smiths who make functional longsword sized blades. I have now (see above comment), and felt comfortable with that as sufficient burden of proof on adding them. If that's not enough, let me know what would be sufficient data or proof! :)

Setting that aside though, at minimum the gimmicky titanium swords are specifically important to me, in selling that titanium can also be a buzzword to sell junky decorative pieces and overpriced knives. There's precedent in using junky decorative rainbow-anodized titanium swords off someone's wall to defend yourself, in the same way as players might desperately try to sharpen a sporting saber, or just start clobbering baddies with a fake aluminum sword.

**Comment by Maleclypse** [comment]
https://swordencyclopedia.com/titanium-swords/#:~:text=Titanium%20is%20lightweight%2C%20strong%20and,for%20slashing%20attacks%20and%20blocking.

An article on why titanium swords are for decorative purposes only.

**Comment by gadgetpatch** [comment]
> https://swordencyclopedia.com/titanium-swords/#:~:text=Titanium%20is%20lightweight%2C%20strong%20and,for%20slashing%20attacks%20and%20blocking.
> 
> An article on why titanium swords are for decorative purposes only.

Noted, but I'd trust a group of smiths specialized in making and using titanium blades/tools professionally as a more reliable source than a history blogger who probably is echoing

> https://swordencyclopedia.com/titanium-swords/#:~:text=Titanium%20is%20lightweight%2C%20strong%20and,for%20slashing%20attacks%20and%20blocking.
> 
> An article on why titanium swords are for decorative purposes only.

Noted, and I've seen similar blogs parroting the same points, but I still trust [a small company of craftspeople](https://www.madscienceforge.com/laboratory/pages/en/faq.php) who not only make but are _specialized_ in making titanium blades and tools, more than I'd trust a blogger and history enthusiast. Especially when her claims about edge retention and material properties are flatly wrong (presumably based on CP1 titanium, not anything hardened or alloyed.)

**Comment by ampersand55** [comment]
Titanium not suitable for edged weapons as it's very soft compared to steel.

Titanium grade 5 has a Rockwell C hardness (HRC) of 36 [[1](https://asm.matweb.com/search/SpecificMaterial.asp?bassnum=mtp641)] compared to steel which have a HRC of 55-70 [[2](https://en.wikipedia.org/wiki/List_of_blade_materials)]. 

https://www.bladeforums.com/threads/best-blade-material-hi-carbon-steel-or-titanium.328787/

**Comment by gadgetpatch** [comment]
> Titanium not suitable for edged weapons as it's very soft compared to steel.
> 
> Titanium grade 5 has a Rockwell C hardness (HRC) of 36 [[1](https://asm.matweb.com/search/SpecificMaterial.asp?bassnum=mtp641)] compared to steel which have a HRC of 55-70 [[2](https://en.wikipedia.org/wiki/List_of_blade_materials)].
> 
> https://www.bladeforums.com/threads/best-blade-material-hi-carbon-steel-or-titanium.328787/

Only half-correct, and that take is based on a common misconception.

1) Edge hardness is convenient, but not _necessary_ for swords and knives. Steel blades from the Early Middle Ages average below 20 Rockwell C, and later pre-modern steels average closer to 40, and never as high as 50. But they still (obviously) worked as swords and armor. The trick? They were sharpened and maintained regularly. We still have to do this with modern carbon steel, just less often. Again, surface hardness is convenient, but not necessary in weapons. (see the [Period Blade Hardness](http://myarmoury.com/feature_bladehardness.html) section, or [this chart](http://myarmoury.com/images/features/pic_mow_bladehardness01.jpg) here).

2) Cataclysm weapon crafting is modelled using the following carbon steels:
-- **"Low Carbon" [SAE 1010](https://www.matweb.com/search/DataSheet.aspx?MatGUID=025d4a04c2c640c9b0eaaef28318d761)**: It's Rockwell C is less than 0. It has to be measured on the soft metal scale instead.
-- **"Mid Carbon" [SAE 1050](https://www.matweb.com/search/DataSheet.aspx?MatGUID=cd45f104b44b4f9e9ebab62e8c127b05)**: Rockwell C ranges from <0 to <20. This is our bog-standard steel, which most things are made from, and is what most modern swordsmiths use for their projects.
-- **"High Carbon" [SAE 1080](https://www.matweb.com/search/DataSheet.aspx?MatGUID=7921010ea331448e82b0194df9dbdf75)**. Rockwell C ranges from <10 to <30. This is good, performance carbon steel.
-- **"Tempered High Carbon" [Tempered SAE 1080](https://www.matweb.com/search/DataSheet.aspx?MatGUID=3c23928592474859bd1950f7c3f8bb01)**  : Rockwell C of ~40. This is our "best of the best" carbon steel that a survivor can manufacture from scrap and charcoal.

Now, for comparison, that forum post you quoted cites _annealed_ Grade 5, which is softened for ease of working. And yet, it still manages a Rockwell C between of 35, between High Carbon and Tempered Steel. Compare this, and steel, with [finished Grade 5 Titanium](https://www.matweb.com/search/DataSheet.aspx?MatGUID=b350a789eda946c6b86a3e4d3c577b39) and you see that it's on part with Tempered Steel at RC 41.

When people cite steel with Rockwell Cs of 50+, those are tool carbides and specialty alloys which survivors can't reasonably craft, and would instead need to scrap them from special sources. Many of them can't practically be repaired or welded, and in order to be superhardened become brittle instead of ductile, losing some of the most useful qualities of a rugged steel weapon. That's why modern swordmakers typically use Mid-Carbon SAE 1060, instead of more exotic steels.

NOTE: I think simulating 50+ hardness steels beyond carbon steel would be very cool, in a similar way to adding Titanium, but that's outside scope for my current work. I'd be open to considering it in future metallurgy work though!

TL;DR, I actually fudged Titanium's stats to be _worse_ for gameplay purposes, because accurately representing it would outshine most steels the player has access to, and could diminish the value of other users' work on carbon steel.

**Comment by gadgetpatch** [comment]
I have more to say on titanium swords, specifically how it's ideal for slicing swords like katanas, or quick stabbing swords like sabers, and likely less good for chopping swords without modifications, but!

To keep this PR moving, I'm going to migrate anything longer than a titanium knife (even the gimmicky decorative katanas) to a future pull request, and balance the numbers down to where I want them.

**Comment by kevingranade** [comment]
For your future reference,  we discussed the madscienceforge swords here https://discord.com/channels/598523535169945603/598535827169083403/1142266321975922829

Tl;dr you have a source claiming to make a superior sword with an unobtainable alloy, but when you scale it down to using an obtainable alloy,  you're still leaving it with superior stats.
In short,  you can meet the requirements of titanium being findable,  workable,  and providing good stats,  but not all at once.

File: data/json/items/melee/knives_kitchen.json [hunk 1]
```diff
@@ -65,6 +117,25 @@
...
+    "ascii_picture": "knife_paring",
+    "copy-from": "base_titanium_kitchen_knife",
+    "looks_like": "knife_paring",
+    "weight": "28 g",
+    "volume": "30 ml",
+    "longest_side": "15 cm",
+    "price": 500,
+    "price_postapoc": 100,
+    "to_hit": { "grip": "weapon", "length": "hand", "surface": "point", "balance": "neutral" },
+    "qualities": [ [ "CUT", 2 ], [ "CUT_FINE", 2 ], [ "BUTCHER", 12 ] ],
```
Comment (TheShadowFerret) [inline data/json/items/melee/knives_kitchen.json:135]:
```suggestion
    "qualities": [ [ "CUT", 2 ], [ "CUT_FINE", 1 ], [ "BUTCHER", 12 ] ],
```
Why would the blade material change its fine cutting :thinking:

File: data/json/items/melee/knives_kitchen.json [hunk 2]
```diff
@@ -125,6 +236,25 @@
     "flags": [ "SHEATH_KNIFE", "ALLOWS_BODY_BLOCK" ],
     "melee_damage": { "bash": 4, "cut": 6 }
   },
+  {
+    "id": "knife_bread_ti",
+    "name": { "str": "titanium bread knife", "str_pl": "titanium bread knives" },
+    "type": "TOOL",
+    "category": "weapons",
+    "description": "This light, performance knife has a long serrated blade for cutting soft bread without squishing it.  It's scalloped teeth can slice and tear well enough in a fight.  Features a wear-resistant titanium alloy blade, durable construction, and even an antimicrobial handle.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/knives_kitchen.json:244]:
```suggestion
    "description": "This light, performance knife has a long serrated blade for cutting soft bread without squishing it.  Its scalloped teeth can slice and tear well enough in a fight.  Features a wear-resistant titanium alloy blade, durable construction, and even an antimicrobial handle.",
```

File: data/json/items/melee/knives_kitchen.json [hunk 3]
```diff
@@ -146,6 +276,25 @@
     "melee_damage": { "bash": 3, "cut": 17 },
     "//2": "Based on https://www.amazon.com/dp/B001JL4J0G"
   },
+  {
+    "id": "knife_vegetable_cleaver_ti",
+    "name": { "str": "titanium vegetable cleaver" },
+    "type": "TOOL",
+    "category": "weapons",
+    "description": "A menacing performance knife with a broad, square chopping blade.  It's size and speed makes it a decent cutting weapon in a pinch.  Features a wear-resistant titanium alloy blade, durable construction, and even an antimicrobial handle.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/knives_kitchen.json:284]:
```suggestion
    "description": "A menacing performance knife with a broad, square chopping blade.  Its size and speed makes it a decent cutting weapon in a pinch.  Features a wear-resistant titanium alloy blade, durable construction, and even an antimicrobial handle.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 4]
```diff
@@ -322,6 +480,59 @@
...
+    "price": 50000,
+    "price_postapoc": 3000,
+    "delete": { "flags": [ "CONDUCTIVE" ] },
+    "gunmod_data": {
+      "location": "underbarrel",
+      "mod_targets": [ "shotgun", "rifle", "crossbow", "launcher" ],
+      "mode_modifier": [ [ "REACH", "bayonet", 2, [ "MELEE", "REACH_ATTACK" ] ] ],
+      "install_time": "5 s"
+    },
+    "qualities": [ [ "CUT", 1 ], [ "CUT_FINE", 1 ], [ "BUTCHER", 20 ] ],
```
Comment (TheShadowFerret) [inline data/json/items/melee/swords_and_blades.json:532]:
```suggestion
    "qualities": [ [ "CUT", 1 ], [ "CUT_FINE", 1 ], [ "BUTCHER", 20 ], [ "DIG", 1 ] ],
```
"digging through soil without setting off magnetic mines"

File: data/json/items/melee/swords_and_blades.json [hunk 5]
```diff
@@ -1630,6 +1859,153 @@
...
+    "proportional": { "melee_damage": { "bash": 8.5, "cut": 0.23 }, "price": 0.1, "price_postapoc": 0.1 },
+    "delete": { "flags": [ "DURABLE_MELEE" ] }
+  },
+  {
+    "id": "ti_katana",
+    "copy-from": "katana",
+    "looks_like": "katana",
+    "type": "TOOL",
+    "name": { "str": "titanium katana" },
+    "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Sturdily built, and quick to swing thanks to it's light weight, it should be quite effective in a fight.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/swords_and_blades.json:1966]:
```suggestion
    "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Sturdily built, and quick to swing thanks to its light weight, it should be quite effective in a fight.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 6]
```diff
@@ -1630,6 +1859,153 @@
...
+    "price_postapoc": 5000,
+    "color": "magenta",
+    "delete": { "flags": [ "CONDUCTIVE" ] },
+    "replace_materials": { "steel": "titanium", "leather": "nylon" },
+    "variant_type": "generic",
+    "variants": [
+      {
+        "id": "ti_katana_generic",
+        "name": { "str": "titanium katana" },
+        "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Sturdily built, and quick to swing thanks to it's light weight, it should be quite effective in a fight.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/swords_and_blades.json:1978]:
```suggestion
        "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Sturdily built, and quick to swing thanks to its light weight, it should be quite effective in a fight.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 7]
```diff
@@ -1630,6 +1859,153 @@
...
+        "weight": 1
+      },
+      {
+        "id": "ti_katana_gradient",
+        "name": { "str": "titanium katana" },
+        "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Its blade is heat-anodized with a vivid purple-to-yellow sunset gradient, from edge to center.  Sturdily built, and quick to swing thanks to it's light weight.  Despite likely having been made for show, it should still be quite effective in a fight.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/swords_and_blades.json:1984]:
```suggestion
        "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Its blade is heat-anodized with a vivid purple-to-yellow sunset gradient, from edge to center.  Sturdily built, and quick to swing thanks to its light weight.  Despite likely having been made for show, it should still be quite effective in a fight.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 8]
```diff
@@ -1630,6 +1859,153 @@
...
+        "color": "magenta",
+        "weight": 2
+      },
+      {
+        "id": "ti_katana_dark",
+        "name": { "str": "titanium katana" },
+        "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Its blade is coated with black carbide, giving it a sleek and futuristic look.  Sturdily built, and quick to swing thanks to it's light weight.  Despite likely having been made for show, it should still be quite effective in a fight.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/swords_and_blades.json:1991]:
```suggestion
        "description": "A long, single-edged weapon patterned after the well-known samurai sword, but made from modern titanium.  Its blade is coated with black carbide, giving it a sleek and futuristic look.  Sturdily built, and quick to swing thanks to its light weight.  Despite likely having been made for show, it should still be quite effective in a fight.",
```

**Comment by gadgetpatch** [comment]
> For your future reference, we discussed the madscienceforge swords here https://discord.com/channels/598523535169945603/598535827169083403/1142266321975922829
>
> Tl;dr you have a source claiming to make a superior sword with an unobtainable alloy, but when you scale it down to using an obtainable alloy, you're still leaving it with superior stats.
In short, you can meet the requirements of titanium being findable, workable, and providing good stats, but not all at once.

In brief: None of these alloys are remotely unobtainable, Grade 5 is in fact the most common alloy in industrial use (that's why I picked it as our standard), and Grade 5 is equal to carbon steel (as I mentioned in the post prior to yours). I'll substantiate all that in a second, and hopefully clear up the misunderstanding.

Also, I want to say, I'm still fine keeping titanium swords to another PR... but if you're bringing this up because you're still open to adding them back in after I've proven the case, I'd be happy to. Either way, I agree that I'd like to tune Grade 5 to be less damaging per strike than tempered steel though. The current damage values are entirely a WIP; I wasn't expecting the lower stamina drain to have such a big impact on DPS!

And don't worry, I remember our conversation. I brought up Sam Taylor's forge so that other people can follow along, and also to remind you. I'm guessing you missed the reply I @'d you with ([discord link](https://discord.com/channels/598523535169945603/598535827169083403/1142267186866552842)). I think there might've been some confusion from our conversation over that FAQ too, so I can try to clear it up real quick.

But before I get to that, I want to share a proper San Mai titanium sword, made of the common industry-standard titanium I'm modelling in this PR. You can see it cutting as cleanly and powerfully as any high grade carbon steel sword... and (in my opinion) looking very pretty doing it!
- [Youtube video](https://www.youtube.com/watch?v=fWqdVtXNb2M&t=157s) (Rewind to see the build and heat treatment, and another alloy sword)

This sword was a collaboration between champion swordsman and smith Dan Keffler, and Sam Taylor of Mad Science Forge. It uses bog-standard US Grade 5 ((Ti-6Al-4V), and Russian VT23. They're the same category of titanium, with very similar composition. VT23 is essentially the Russian version Grade 5, adding <1% Stainless Steel, making it about as easy to forge as steel, with little loss of blade or armor performance. This army study provides a comparative analysis, to back that claim up: [PDF Link](https://apps.dtic.mil/sti/pdfs/ADA420984.pdf).

Long story short, Sam and Dan fold Grade 5 and VT23 together in their "multi-alloy swords" to make the titanium a bit easier to hammer and temper. It's not necessary for blade performance, but the ease-of-working improves the final build quality, without compromising it's other properties. All of these alloys (including Ti-NB45, the "aircraft alloy" they mention in their FAQ) are very available too. They were "from the 80s and 90's" in that they were _invented_ then, not to say that's the last time they were made. You or I can still buy them from ATI in Texas, Elektrovek Steel in the EU, etc.

Also, in response to your worries that Grade 5 is worse than carbon steel, I offered a quick comparison of the properties of both in [the comment immediately before your post](https://github.com/CleverRaven/Cataclysm-DDA/pull/67991#issuecomment-1704344958), and some data on how it compares to recovered medieval swords too. That's why I still feel confident modelling Grade 5 and Carbon Steel to have similar damage-per-hit... though I still may want to make them worse than they really are, for balance reasons.

Last thing, but I clipped a section of Sam Taylor testing one of his alloy swords on the show Knife or Death. The whole episode is a fantastic comparison of carbon steel and titanium alloy as a sword metal.
- [This short clip](https://drive.google.com/file/d/1i3NaibdmbCcHhyX0Fm-iUAwnRj3pZ1Cu/view?usp=sharing) shows his sword completely dominating their sword testing gauntlet.
- [This is a longer clip](https://drive.google.com/file/d/1ERNixHm-80MNRBggetMZNcvKa6jRzvwJ/view?usp=drive_link), showing how it compares to a carbon steel broadsword, seax, kukri, and falcata.

File: data/json/items/melee/knives_kitchen.json (hunk 1 above)
Comment (gadgetpatch) [reply data/json/items/melee/knives_kitchen.json:135]: Paring knives in general are designed for controlled small cuts, in a way standard knives aren't (they're much more like an x-acto knife in that way). I'd like to update steel paring knives with fine cutting 2 as well, but I was going to handle that separately. Thoughts?

File: data/json/items/melee/swords_and_blades.json (hunk 4 above)
Comment (gadgetpatch) [reply data/json/items/melee/swords_and_blades.json:532]: Thanks for catching this! I had dig 1 on it for my first draft of this PR, but must've lost it while refactoring and not noticed.

P.S., thanks for catching all the Its/It's. Trips me up all the time! ><;

File: data/json/items/melee/knives_kitchen.json (hunk 1 above)
Comment (gadgetpatch) [reply data/json/items/melee/knives_kitchen.json]: Just went ahead and removed Fine Cutting 2 for now, and will consider adding it to all paring knives in separate PR.

**Comment by gadgetpatch** [comment]
@kevingranade Just as a quick recap:
- I took out the ballistic side plates, and some of the cosmetic item variants, and will put them in another PR.
- Titanium blades have been tuned down in damage per hit, to match (or be slightly worse) than equivalent carbon steel.
- Titanium sword definitions remain, but they don't exist in any spawn groups. Will deal with that in a future PR.

**Comment by gadgetpatch** [comment]
Update: Went ahead and resolved a potential merge conflict with the tool itemgroup, that's emerged since this PR was made.

Just trying to help make this as low-friction as possible where I can.

**Comment by Brambor** [comment]
You could remove the WIP in title since this is ready for Kevin's blessing. Then merge.

**Review by Maleclypse — CHANGES_REQUESTED** [review]
I came here tonight with the plan to review this. I'm going to try and give you concrete feedback with direct steps.   
1. It's too large. The maximum size for non Mapgen JSON PRs is 2k.  The ideal is 500 lines.  Your first file in the item groups has 513 lines in it with probably only a couple dozen of them relating to titanium.  
2. Divide this PR into multiple PRs.  First I'd suggest a couple of item PRs.  At minimum one for armor, one for weapons and one for the remaining items.   Then one that handles item group changes and monster drops.  Preferably only adding the new items.  The non titanium item changes should be their own PR for ease of review.   
3. Review your work for coherency.  When I started looking at the knives the first two "folding_knife" and "diveknife" both reduce the weight of the item by almost half it's weight and then add additional bash damage.  You state it's coming from the hilt but folding knives don't do damage from the hilt.  Stab weapons probably shouldn't ever do bash damage which isn't your problem but looking at this PR made it visible.   I started writing up multiple questions but eventually realized I'd be here all night or it would take so long I'd never finish it.  Reviewer fatigue is your enemy, see point one.  It's also impossible for anyone to feel confident that all of their questions have been answered when there is so much content to review and it breeds the fear that if you approve this you are probably missing things you would have asked about but lost sight of because it's just so large. 


  I've pulled a copy of the PR down to my machine and can try to help you divide it up if you need help.  Please ping or DM me on discord if you want me to help divide this up.

**Comment by gadgetpatch** [comment]
@Maleclypse, I appreciate you stepping in and taking the time to explain what you assume @kevingranade's problem was. I wish Kevin had been responsive, followed up, or even explained himself, since it is his review holding this PR hostage.

> I came here tonight with the plan to review this. I'm going to try and give you concrete feedback with direct steps.
> 1. It's too large. The maximum size for non Mapgen JSON PRs is 2k. The ideal is 500 lines. Your first file in the item groups has 513 lines in it with probably only a couple dozen of them relating to titanium.
>
> 2. Divide this PR into multiple PRs.  First I'd suggest a couple of item PRs.  At minimum one for armor, one for weapons and one for the remaining items.   Then one that handles item group changes and monster drops.  Preferably only adding the new items.  The non titanium item changes should be their own PR for ease of review.
>
> 3. Review your work for coherency.  (snip)

For points 1 and 2, these are understandable requests; if these are meant to be general guidelines for the project, they'd have been great to know before I started, and should be documented somewhere (and agreed on by existing leads). Note that I received opposite instruction from other devs on the discord during development (more on this below).

For issue 3, as you say it isn't my issue, and is beyond the scope of this PR. I was following the pattern established by existing C:DDA content. In reality, most weapons can strike in many ways, so I assumed multiple damage types were meant to abstract possible pommel strikes, stabs, slashes, etc. If weapons are not meant to do this, and damage represents a single "proper" way to use a weapon, consider instead auditing C:DDA for coherency, and creating documentation to prevent this sort of confusion in future.

### 1) If you aren't able to review the PR, then it doesn't make sense for you (or Kevin) to personally request to review it, then complain it's too much for you to review, does it?
### 2) It doesn't make sense to review this PR for following a pattern in established core content. Your personal feeling that it should be changed should be an issue for a future PR.

I was given *direct* instruction by several established devs on the discord that the PR would not be reviewed if I included only raw material definitions; that there was no reason to submit it unless I also included a body of immediately usable weapons and armor content (with drop groups to match). I did so. Now, you are asking the opposite; that I remove any usable content for ease of review.

This represents hours of work, to undo hours of work done at other devs' requests; which is a pattern that's repeated itself since early in this PR's development, which I've done my best to satisfy until this point. I am no longer interested in doing so.

-----

This conflicting direction follows an established pattern. I was told directly by Kevin that he (personally, for some reason) would only accept the addition of Titanium if it was better in every way than Steel, as that was "the only reason to consider adding it". I was then later told the opposite by Kevin; that unless it was worse than steel in every way, it wouldn't make sense and would be rejected. Obviously, I cannot satisfy both of these requirements; he is in conflict with himself.

Despite that I respect his systems programming chops, I know that Kevin's game design intuition is poor, and that his behavior as "project lead" is a known problem... one that's splintered this project before, and driven away countless developers. I've seen his unprofessionalism and hostility, his tendency to create conflicts in bad faith, and his authoritarian tendency to throw his weight around despite this being an ostensibly open-source project.

I also understand (from several DMs on the project Discord apologizing for Kevin's behavior, a few encouraging me not to leave and to just tough it out, and several open statements in this repo) that existing developers treat dealing with Kevin as a cost of contributing to the project. So perhaps Kevin coming into this PR, freezing it with an issue that'd already been resolved, getting confused and demanding explanation, then vanishing and leaving the PR in limbo without qualifying or following up once, and leaving another dev to tidy up after him, is just something contributors are expected to deal with?

Personally, I have no interest in quietly tolerating this on an ongoing basis. I have little patience for wasted work and conflicting direction, let alone FOSS leads being this irresponsible and shitty, and playing petty despot for fun.

### 3) I cannot satisfy the conflicting demands made by various leads, some of whom have threatened to can the PR if their personal wishes aren't met, and some of which are in conflict with other devs demanding the opposite. I'm not interested in gambling my time and effort on conflicting direction like this.
### 4) Kevin Grande's penchant for hazing particular devs, and his general unprofessional behavior as a project lead, is an ongoing, unresolved issue. 
### 5) Lastly, I've learned that one of the project leads is selling our collective work on C:DDA for their personal profit on Steam. That this is quietly tolerated, without pushback from other leads, is deeply disappointing, and unacceptable to me.

-----

With this said, I will be closing this PR. At this time, I do not intend to contribute directly to C:DDA content or code, and any further contributions will be external development tools applicable to this (and other) branches of Cataclysm.

I am sorry if this comes as a disappointment to any of the people looking forward to the inclusion of this. If anyone is interested in using this work in the future, they are welcome to adapt it.

**Comment by GuardianDll** [comment]
I don't know anything about titanum and not particulary interested in this pr, but what i don't like is how you depict the contributor team, so here is how i see it:

1. It is not a strict rule, but it's size drastically increases the difficulty of the PR to be reviewable, and it's not always about the string size, there is examples when the change has giant amount of strings, but the purpose of the PR is one, single change. Dev team told you to have all important stuff in initial PR, but it didn't tell you to include all related changes in one single PR - they are not dumb and do not want to shoot themselves in the foot
2.  #68190 is a great example, it's a PR that added the huge amount of related, but different parts of the same mechanics. Same as you, they was told to divine it because it's just difficult to review it, and they did it - win win situation for both contributor and for reviewer
3. "Frequently Made Suggestions", the very first page: `The method for users to influence development is via debate. This is intentional, not due to laziness.`. Yes, people do have different opinion in this game, since there is more than one person who contribute to this project. Kevin didn't agree with the pr idea, but he didn't close it either, just requested the changes (ironically the second change request from Maleclypse did nothing but repeat the first point of Kevin's message, just with more details)
4. Kevin is not the easiest person to talk with, but his opinion for me is always weighted and reasonable, and the cases where Kevin itself review the PR are ocasional. This change was one of them, as it touches pretty difficult theme lot of people do not or misunderstand
5. Our license specifically allows this, it could be absolutely anyone - Korg was just a person willing to spend their time to do it; It doesn't affect the game in any possible way whatsoever

This message is not adressed particularly to you, but to anyone who will read this PR, so they could undestand the reality is not completely equal to what you say

**Comment by Zireael07** [comment]
Re: 4) unfortunately while Kevin is usually reasonable there are occasional outbursts (for lack of a better word), and I think there's been more of those recently...

That said, he didn't close the idea outright as usually happens when people start on something that TOTALLY doesn't fit the game's vision, so I guess this would be fine to continue....

**Comment by gadgetpatch** [comment]
You're welcome to disagree with my opinions or perspective, but please don't get me wrong. With the exception of my every interaction with Kevin, I've had great experiences with every other senior and junior developer, and many have been nothing but helpful and supportive since I started contributing. Maleclypse is obviously only trying to help too. That's part of what made this decision difficult, and why I've been giving it time and consideration.

This was just an explanation of my reasons for closing this PR, and withdrawing from direct contributions to this project for the foreseeable future. I recognize that several factors have made working on this branch of Cataclysm unpleasant and not worth it to me, personally. I also recognize that there's no avenue for me to try to improve conditions for myself or others (or I would try to).

That said, out of respect for the work Kevin does do on this project (whatever my feelings toward him may be), I'd prefer this closed PR doesn't turn into a discussion thread. I've said what I have to say.

**Comment by esotericist** [comment]
if folks want to discuss the events that took place here, there are venues for it, but closed PRs are generally not the place for this sort of discussion.

let's try to move on from here.

**Comment by KeremBabaG** [comment]
can we add this to abandoned but desired prs? This is a good one, would be shame if it went to shitter because of kevin or whatever

**Comment by esotericist** [comment]
if someone thinks they can attempt to approach the topic in a fashion that will meet the necessary requirements (including convincing kevin), it's not forbidden. but understand that the hill is steep.

locking this to replies because closed PRs are not the place for people to be throwing rocks. (or really the place for much of anything that isn't actively trying to make progress. this last comment was not that.)
